portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,6 +41,7 @@
 public class DataServlet extends HttpServlet {
 
   private int nextUniqueId;
+  private int currentIndexOfFirstComment;
```

Comment:
is it possible to make this a variable that only lives within a method, and not maintained throughout the class? I'm having difficulty tracking it down

Reply:
I've looked through my code and I'm having trouble figuring out how I could create this within a method. It keeps track of which comment is currently the first comment being displayed (i.e. the "head" of this page of comments). Because it tracks this between get requests to /data (and I can't currently think of a simple way to retrieve this information from the page), it needs to be preserved between function calls to doGet. I don't know a way to preserve the value of a variable declared within a method between method calls. Is there a way to do this? If not, is there a better name for this variable, or documentation I could add that would make it easier to keep track of what this variable is doing? I agree that global variables like this make the code more confusing and I want it to be as readable as possible.

Reply:
I believe the global variable here is shared among different sessions. Meaning if one user changed the current index, all others are affected.

Usually pagination is implemented by passing a page token (where page starts) and a page size (can have a default size) from the request to the server. In this case, javascript can calculate the page token based on page number and pass it to the servlet.

Reply:
Oh I hadn’t thought of it like that!! I’ll be sure to fix that tomorrow. Just to be sure I’m thinking of this correctly: if I create a global variable in my JavaScript file (whether it’s the head comment index or page number), will that also be shared among all sessions? Should I instead create a hidden element in my HTML to keep track of this index to be sure that every user has their own value? I want to be sure I’m choosing a way to keep track of this value (so that I can pass it to the server) while avoiding the issue of it being shared. Thanks!